### PR TITLE
Fix report display same infra for all test plans

### LINKED ISSUE #115: Generate infrastructure test plan automatically instead of hard-coding all the combinations.
**description**
Generate infrastructure test plan automatically instead of hard-coding all the combinations.

### LINKED ISSUE #188: Final report displays same infra information for all the test plans
**Description:**
Final report displays same infra information for all the test plans. Refer [1] for such a report.

[1] http://ec2-34-238-28-168.compute-1.amazonaws.com:8080/job/wso2is-5.4.0/job/01-single-node-deployment-pattern/156/HTML_Report/wso2is-5.3.0-2017-11-30%2013%3A31%3A25.0.html

### PR DESCRIPTION
**Pull request by vidurananayakkara**
## Purpose
Resolves #188 
Resolves #115 

## Release note
* Fix report display same infra for all test plans
* Generate infrastructure test plan automatically instead of hard-coding all the combinations

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java [hunk 1]
```diff
@@ -22,124 +22,237 @@
...
-import java.sql.Timestamp;
-import java.util.Date;
+import java.util.Optional;
 
 
 /**
  *
  */
-public class RunTestPlanCommand extends Command {
+public class RunTestPlanCommand implements Command {
```
Comment (azinneera) [inline core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java]: Unnecessary new line

File: core/src/main/java/org/wso2/testgrid/core/util/GitUtil.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import java.io.File;
+import java.nio.file.Paths;
+
+/**
+ * Util class to handle git related activities.
+ *
+ * @since 1.0.0
+ */
+public class GitUtil {
+
```
Comment (azinneera) [inline core/src/main/java/org/wso2/testgrid/core/util/GitUtil.java]: Unwanted new line

File: core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java [hunk 3]
```diff
@@ -22,124 +22,237 @@
...
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.sql.Timestamp;
-import java.util.Date;
+import java.util.Optional;
 
 
 /**
  *
  */
```
Comment (sameerawickramasekara) [inline core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java]: Shall we add a class comment here

File: core/src/main/java/org/wso2/testgrid/core/util/GitUtil.java (hunk 2 above)
Comment (vidurananayakkara) [reply core/src/main/java/org/wso2/testgrid/core/util/GitUtil.java]: Kept the blank line for code readability

**Pull request merged**